### ISSUE TITLE
Lazy Loading of Services and Types in Ads Output

### ISSUE THREAD

**Issue by BenRKarl** [googleapis/gapic-generator-python#298]
The current version of our client library uses the `__init__.py` for each API version to lazy load client modules, see [here](https://github.com/googleads/google-ads-python/blob/master/google/ads/google_ads/v2/__init__.py).

The current generated `__init__.py` file appears to have a few flaws:

1. The map of classes/filenames is incomplete, right now it only has:

```
_lazy_name_to_package_map = {
    'types': 'google.ads.google_ads_v2.types',
    'enums': 'google.ads.google_ads_v2.enums',
}
```

If I add:
 
```
'google_ads_service_client': 'google.ads.google_ads.v2.services.services.google_ads_service'
```
to the map it can import the google_ads_service_client module when `'GoogleAdsServiceClient'` is passed to the `__getattr__` method, but the import fails to complete for two reasons:

1. The import paths don't match the client library structure. The generated code attempts to use

```
from google.ads.google_ads_v2 ... import xyz
```

But the library is structured such that these statements should be:

```
google.ads.google_ads.v2 ... import xyz
```

I can get around this by using `sed` to find/replace but it would be nice if the micro generator output this structure on its own. This problem is summarized in https://github.com/googleapis/gapic-generator-python/issues/297.

2. There are a number of undefined variables in the files that get imported along with `google_ads_service_client` as described in https://github.com/googleapis/gapic-generator-python/issues/296.

**Comment by hkdevandla** [comment]
Assigning this to Dov since he is already looking into it.

**Comment by software-dov** [comment]
The first part is an artefact of the ads proto directory structure: other submodules' `__init__.py` files show the subpackages and types expanded out. The fix is a missing feature tracked by #304 

The second part is tracked by #297 

Closing this issue as a duplicate of two existing issues.
